### PR TITLE
CF-137 Set generate load use ssh-agent

### PR DESCRIPTION
**Pull request by ntymtsiv**

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:15: [W0403(relative-import), ] Relative import 'base', should be 'tests.base'
devlab/tests/utils.py:19: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:35: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:37: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:131: [E1129(not-context-manager), MigrationUtils.start_ssh_agent] Context manager 'generator' doesn't implement __enter__ and __exit__.
devlab/tests/utils.py:152: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/utils.py [hunk 1]
```diff
@@ -123,6 +124,21 @@ class MigrationUtils(object):
...
+    def start_ssh_agent():
+        with hide('everything'), settings(warn_only=True):
+            result = local('ssh-add -l')
+            if result.succeeded:
+                # Add default rsa key to the agent. Key won't be added second
+                # time (in case when key already added)
+                local('ssh-add')
+                return
+            cmd = 'eval `ssh-agent` && echo $SSH_AUTH_SOCK && ssh-add'
+            info_agent = local(cmd, capture=True).split("\n")
```
Comment (roman-verchikov) [inline devlab/tests/utils.py]: `splitlines()`?

File: devlab/tests/utils.py [hunk 2]
```diff
@@ -123,6 +124,21 @@ class MigrationUtils(object):
...
+            os.environ["SSH_AGENT_PID"] = info_agent[0].split(" ")[-1]
```
Comment (roman-verchikov) [inline devlab/tests/utils.py]: Please use regex

Comment (roman-verchikov) [inline devlab/tests/utils.py]: Actually will it work without setting environment variables? Maybe these two lines are not needed?..

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module docs/conf.py
docs/conf.py:1: [F0001(fatal), ] No module named docs/conf.py
************* Module cloudferrylib.scheduler.scheduler
cloudferrylib/scheduler/scheduler.py:25: [W0403(relative-import), ] Relative import 'cursor', should be 'cloudferrylib.scheduler.cursor'
cloudferrylib/scheduler/scheduler.py:26: [W0403(relative-import), ] Relative import 'signal_handler', should be 'cloudferrylib.scheduler.signal_handler'
cloudferrylib/scheduler/scheduler.py:27: [W0403(relative-import), ] Relative import 'task', should be 'cloudferrylib.scheduler.task'
cloudferrylib/scheduler/scheduler.py:28: [W0403(relative-import), ] Relative import 'thread_tasks', should be 'cloudferrylib.scheduler.thread_tasks'
cloudferrylib/scheduler/scheduler.py:54: [W0613(unused-argument), BaseScheduler.event_end_task] Unused argument 'task'
cloudferrylib/scheduler/scheduler.py:58: [W0613(unused-argument), BaseScheduler.event_error_task] Unused argument 'task'
cloudferrylib/scheduler/scheduler.py:58: [W0613(unused-argument), BaseScheduler.event_error_task] Unused argument 'e'
cloudferrylib/scheduler/scheduler.py:76: [W0703(broad-except), BaseScheduler.process_chain] Catching too general exception Exception
cloudferrylib/scheduler/scheduler.py:85: [W0201(attribute-defined-outside-init), BaseScheduler.process_chain] Attribute 'exception' defined outside __init__
cloudferrylib/scheduler/scheduler.py:108: [W0201(attribute-defined-outside-init), BaseScheduler.addCursor] Attribute 'cursor' defined outside __init__
************* Module tests.testcases.rollback_verification
devlab/tests/testcases/rollback_verification.py:27: [E0611(no-name-in-module), ] No name 'config' in module 'tests'
devlab/tests/testcases/rollback_verification.py:27: [E0401(import-error), ] Unable to import 'tests.config'
devlab/tests/testcases/rollback_verification.py:28: [E0611(no-name-in-module), ] No name 'functional_test' in module 'tests'
devlab/tests/testcases/rollback_verification.py:28: [E0401(import-error), ] Unable to import 'tests.functional_test'
devlab/tests/testcases/rollback_verification.py:29: [E0611(no-name-in-module), ] No name 'utils' in module 'tests'
devlab/tests/testcases/rollback_verification.py:29: [E0401(import-error), ] Unable to import 'tests.utils'
devlab/tests/testcases/rollback_verification.py:30: [E0611(no-name-in-module), ] No name 'data_collector' in module 'tests'
devlab/tests/testcases/rollback_verification.py:30: [E0401(import-error), ] Unable to import 'tests.data_collector'
devlab/tests/testcases/rollback_verification.py:48: [W0201(attribute-defined-outside-init), RollbackVerification.test_verify_rollback] Attribute 'maxDiff' defined outside __init__
************* Module tests.functional_test
devlab/tests/functional_test.py:15: [E0401(import-error), ] Unable to import 'config'
devlab/tests/functional_test.py:18: [E0401(import-error), ] Unable to import 'generate_load'
devlab/tests/functional_test.py:19: [E0401(import-error), ] Unable to import 'utils'
devlab/tests/functional_test.py:21: [E0401(import-error), ] Unable to import 'test_exceptions'
devlab/tests/functional_test.py:180: [W0106(expression-not-assigned), FunctionalTest.filter_volumes] Expression "[volumes.extend(i['cinder_volumes']) for i in config.tenants if ('cinder_volumes' in i) and (not i.get('deleted'))]" is assigned to nothing
devlab/tests/functional_test.py:230: [W0106(expression-not-assigned), FunctionalTest.get_vms_with_fip_associated] Expression "[vms.extend(i['vms']) for i in config.tenants if 'vms' in i]" is assigned to nothing
************* Module tests.testcases.test_verify_dst_del_tenant_resources
devlab/tests/testcases/test_verify_dst_del_tenant_resources.py:16: [E0611(no-name-in-module), ] No name 'config' in module 'tests'
devlab/tests/testcases/test_verify_dst_del_tenant_resources.py:16: [E0401(import-error), ] Unable to import 'tests.config'
devlab/tests/testcases/test_verify_dst_del_tenant_resources.py:17: [E0611(no-name-in-module), ] No name 'functional_test' in module 'tests'
devlab/tests/testcases/test_verify_dst_del_tenant_resources.py:17: [E0401(import-error), ] Unable to import 'tests.functional_test'
************* Module tests.testcases.test_verify_dst_net_segm_id
devlab/tests/testcases/test_verify_dst_net_segm_id.py:15: [E0611(no-name-in-module), ] No name 'functional_test' in module 'tests'
devlab/tests/testcases/test_verify_dst_net_segm_id.py:15: [E0401(import-error), ] Unable to import 'tests.functional_test'
************* Module tests.testcases.test_group_verification
devlab/tests/testcases/test_group_verification.py:66: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\d'. String constant might be missing an r prefix.
devlab/tests/testcases/test_group_verification.py:66: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\.'. String constant might be missing an r prefix.
devlab/tests/testcases/test_group_verification.py:66: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\d'. String constant might be missing an r prefix.
devlab/tests/testcases/test_group_verification.py:66: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\.'. String constant might be missing an r prefix.
devlab/tests/testcases/test_group_verification.py:66: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\d'. String constant might be missing an r prefix.
devlab/tests/testcases/test_group_verification.py:66: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\.'. String constant might be missing an r prefix.
devlab/tests/testcases/test_group_verification.py:66: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\d'. String constant might be missing an r prefix.
devlab/tests/testcases/test_group_verification.py:21: [E0611(no-name-in-module), ] No name 'functional_test' in module 'tests'
devlab/tests/testcases/test_group_verification.py:21: [E0401(import-error), ] Unable to import 'tests.functional_test'
devlab/tests/testcases/test_group_verification.py:61: [W0703(broad-except), GroupProcedureVerification.tearDown] Catching too general exception Exception
devlab/tests/testcases/test_group_verification.py:97: [W0201(attribute-defined-outside-init), GroupProcedureVerification._prepare_files] Attribute 'new_file_name' defined outside __init__
devlab/tests/testcases/test_group_verification.py:74: [W0201(attribute-defined-outside-init), GroupProcedureVerification._prepare_files] Attribute 'pre_conf_dict' defined outside __init__
devlab/tests/testcases/test_group_verification.py:80: [W0201(attribute-defined-outside-init), GroupProcedureVerification._prepare_files] Attribute 'pre_conf_dict' defined outside __init__
devlab/tests/testcases/test_group_verification.py:73: [W0201(attribute-defined-outside-init), GroupProcedureVerification._prepare_files] Attribute 'post_conf_dict' defined outside __init__
devlab/tests/testcases/test_group_verification.py:108: [W0201(attribute-defined-outside-init), GroupProcedureVerification._prepare_files] Attribute 'post_conf_dict' defined outside __init__
devlab/tests/testcases/test_group_verification.py:134: [W0201(attribute-defined-outside-init), GroupProcedureVerification.src_vms_info_generator] Attribute 'src_vms' defined outside __init__
devlab/tests/testcases/test_group_verification.py:186: [W0201(attribute-defined-outside-init), GroupProcedureVerification.test_verify_vm_status_filtering_during_grouping] Attribute 'src_vms' defined outside __init__
************* Module tests.testcases.test_vm_migration
devlab/tests/testcases/test_vm_migration.py:19: [E0611(no-name-in-module), ] No name 'functional_test' in module 'tests'
devlab/tests/testcases/test_vm_migration.py:19: [E0401(import-error), ] Unable to import 'tests.functional_test'
devlab/tests/testcases/test_vm_migration.py:20: [E0611(no-name-in-module), ] No name 'config' in module 'tests'
devlab/tests/testcases/test_vm_migration.py:20: [E0401(import-error), ] Unable to import 'tests.config'
devlab/tests/testcases/test_vm_migration.py:26: [E0203(access-member-before-definition), VmMigration.setUp] Access to member 'filter_vms' before its definition line 44
************* Module tests.testcases.test_verify_dst_functionality
devlab/tests/testcases/test_verify_dst_functionality.py:15: [E0611(no-name-in-module), ] No name 'config' in module 'tests'
devlab/tests/testcases/test_verify_dst_functionality.py:15: [E0401(import-error), ] Unable to import 'tests.config'
************* Module tests.utils
devlab/tests/utils.py:15: [E0401(import-error), ] Unable to import 'base'
devlab/tests/utils.py:19: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:21: [E0401(import-error), ] Unable to import 'test_exceptions'
devlab/tests/utils.py:37: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:39: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:146: [E1129(not-context-manager), MigrationUtils.start_ssh_agent] Context manager 'generator' doesn't implement __enter__ and __exit__.
devlab/tests/utils.py:165: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.
devlab/tests/utils.py:26: [W0611(unused-import), ] Unused agent imported from paramiko
************* Module tests.testcases.test_resource_migration
devlab/tests/testcases/test_resource_migration.py:15: [E0611(no-name-in-module), ] No name 'config' in module 'tests'
devlab/tests/testcases/test_resource_migration.py:15: [E0401(import-error), ] Unable to import 'tests.config'
devlab/tests/testcases/test_resource_migration.py:16: [E0611(no-name-in-module), ] No name 'test_exceptions' in module 'tests'
devlab/tests/testcases/test_resource_migration.py:16: [E0401(import-error), ] Unable to import 'tests.test_exceptions'
devlab/tests/testcases/test_resource_migration.py:17: [E0611(no-name-in-module), ] No name 'functional_test' in module 'tests'
devlab/tests/testcases/test_resource_migration.py:17: [E0401(import-error), ] Unable to import 'tests.functional_test'
devlab/tests/testcases/test_resource_migration.py:596: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas.get_tenant_quotas] Access to a protected member _info of a client class
devlab/tests/testcases/test_resource_migration.py:606: [W0212(protected-access), ResourceMigrationTests.test_migrate_tenant_quotas] Access to a protected member _info of a client class

**Comment by MirantisWorkloadMobility** [comment]
There were no python files updated. Admins, please, check pull request.

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:15: [W0403(relative-import), ] Relative import 'base', should be 'tests.base'
devlab/tests/utils.py:19: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/utils.py:37: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:39: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:146: [E1129(not-context-manager), MigrationUtils.start_ssh_agent] Context manager 'generator' doesn't implement __enter__ and __exit__.
devlab/tests/utils.py:165: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.
devlab/tests/utils.py:26: [W0611(unused-import), ] Unused agent imported from paramiko

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/2415/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:15: [W0403(relative-import), ] Relative import 'base', should be 'tests.base'
devlab/tests/utils.py:19: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/utils.py:36: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:38: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:145: [E1129(not-context-manager), MigrationUtils.start_ssh_agent] Context manager 'generator' doesn't implement __enter__ and __exit__.
devlab/tests/utils.py:164: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by ntymtsiv** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:15: [W0403(relative-import), ] Relative import 'base', should be 'tests.base'
devlab/tests/utils.py:19: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/utils.py:36: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:38: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:145: [E1129(not-context-manager), MigrationUtils.start_ssh_agent] Context manager 'generator' doesn't implement __enter__ and __exit__.
devlab/tests/utils.py:164: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/utils.py [hunk 3]
```diff
@@ -123,6 +125,32 @@ class MigrationUtils(object):
...
         self.config = conf
+        self.start_ssh_agent()
+
+    def start_ssh_agent(self):
+        if self.config.local_rsa_key_path:
+            rsa_key_path = self.config.loca_rsa_key
+        else:
+            # check default path ~/.ssh/id_rsa and ~/.ssh/id_dsa
+            for key in ('/.ssh/id_rsa', '/.ssh/id_dsa'):
+                path_to_default_key = os.environ['HOME'] + key
```
Comment (roman-verchikov) [inline devlab/tests/utils.py]: `os.path.join`

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils
devlab/tests/utils.py:15: [W0403(relative-import), ] Relative import 'base', should be 'tests.base'
devlab/tests/utils.py:19: [W0403(relative-import), ] Relative import 'config', should be 'tests.config'
devlab/tests/utils.py:21: [W0403(relative-import), ] Relative import 'test_exceptions', should be 'tests.test_exceptions'
devlab/tests/utils.py:36: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:38: [W0141(bad-builtin), convert] Used builtin function 'map'. Using a list comprehension can be clearer.
devlab/tests/utils.py:145: [E1129(not-context-manager), MigrationUtils.start_ssh_agent] Context manager 'generator' doesn't implement __enter__ and __exit__.
devlab/tests/utils.py:164: [E1129(not-context-manager), MigrationUtils.execute_command_on_vm] Context manager 'generator' doesn't implement __enter__ and __exit__.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: devlab/tests/utils.py [hunk 4]
```diff
@@ -123,6 +125,32 @@ class MigrationUtils(object):
 
     def __init__(self, conf):
         self.config = conf
+        self.start_ssh_agent()
+
+    def start_ssh_agent(self):
+        if self.config.local_rsa_key_path:
```
Comment (roman-verchikov) [inline devlab/tests/utils.py]: Might actually be better with `local_rsa_key_path` set to `~/.ssh/id_rsa` by default

**Comment by roman-verchikov** [comment]
closing for now since it's a low priority issue

**Pull request closed without merging**